services/generic/parameters.md

```diff
@@ -0,0 +1,56 @@
+|--------------|--------|--------------------------------------|-------------------------------------------------------------------------------|
+| parameter_id | string | `"1"`                                | Id of the parameter.                                                          |
+| name         | string | `"Sensitivity of the PR sensor"`     | Name of the parameter.                                                        |
+| description  | string | `""`                                 | Description of the parameter.                                                 |
+| widget_type  | string | `"select"`                           | Format of the parameter, one of [`widget_type`](#definitions).                |
+| value_type   | string | `"int"`                              | Type of the value.                                                            |
+| options      | object | `[{"label":"Option 1", "value": 1}]` | Only for`select` and `multiselect`. Array of [`select_option`](#definitions). |
+| min          | int    | `-1`                                 | Only for `input`. Minimum possible value of the parameter.                    |
+| max          | int    | `20`                                 | Only for `input`. Maximum possible value of the parameter.                    |
+| default      | int    | `1`                                  | Default value of the parameter.                                               |
+| read_only    | bool   | `false`                              | If true - value cannot be set.                                                |
+
+* `widget_type` can be one of:
+  * `input` - input a number,
+  * `select` - select one item from supported options,
+  * `multiselect` - select multiple items from supported options.
+  
+* `select_option` is an object of the following structure:
+
+| Field | Type   | Example      | Description          |
+|-------|--------|--------------|----------------------|
+| label | string | `"Option 1"` | Label of the option. |
+| value | int    | `1`          | Value of the option. | 
+
+* `parameter_value` is an object of the following structure:
+
+| Field        | Type                  | Example | Description                                                                                              |
+|--------------|-----------------------|---------|----------------------------------------------------------------------------------------------------------|
+| parameter_id | string                | `"1"`   | Id of the parameter.                                                                                     |
+| value_type   | string                | `"int"` | Type of the value. Can be either`"int"` (for `input` and `select`) or `"int_array"` (for `multiselect`). |
+| value        | based on value_type   | `1`     | Value of the parameter.                                                                                  |
```

Comment:
What would you say if we had strongly typed object with fields:
- value - int
- values - int array

@PaulinaDrozda @BudzikM @eternalamenity 

This way, there is no partial parsing and whatnot? Also since we implement ints only at this stage,  if needed we can extend this object further. Personally, I prefer to avoid undefined types at all costs.

Reply:
Ooo, way better, I think

Reply:
But if we want to have a possibility to extend this further then maybe name the fields in an even more detailed manner, like "int_value" "int_map_value", this way anything more that we add can follow the same principle. What do you think?

Reply:
It's pure aesthetics.

The thing is that if we use int_value here we also need to use int_value in option object in the params specification and int_default. If so, then it's better to already create value object and have it in spec.Default, spec.options.value and command/report as well.

Reply:
Something like this:

Shared object

```
{
"value_type": "int",
"int_value": 1,
...
}
```

Command/Report object:

```
{
"parameter_id": "1",
"value": {
   "value_type": "int",
   "int_value": 1,
}
}
```

And the same in spec:

```
{
...
"default": {"value_type":...}
"options": [
  {
      "label": "x",
      "value": {"value_type"...}
  }
]

}
```